.github/workflows/review_and_testing.yml

```diff
@@ -1,10 +1,33 @@
-name: testing
+name: Review & Test
+
+permissions:
+  contents: read
+  pull-requests: write
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The permissions set for this workflow are `read` for contents and `write` for pull requests. This is a good practice as it follows the principle of least privilege, granting only the necessary permissions for the workflow to function.

```diff
  permissions:
    contents: read
    pull-requests: write
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->